Work around lack of _getshort in resolv.h in Xcode 26.

./dns.c:118:10: error: call to undeclared function '_getshort'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
--- dns.c.orig	2023-09-05 12:50:56.000000000 -0500
+++ dns.c	2025-09-25 21:24:34.000000000 -0500
@@ -51,6 +51,10 @@
 #include <string.h>
 #include <stdio.h>
 
+#if defined(__APPLE__) && !defined(_getshort)
+#define _getshort res_9_getshort
+#endif
+
 #include "gnuc.h"
 #ifdef HAVE_OS_PROTO_H
 #include "os-proto.h"
